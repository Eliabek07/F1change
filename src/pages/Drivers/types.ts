export interface Root {
  MRData: Mrdata;
}

export interface Mrdata {
  xmlns: string;
  series: string;
  url: string;
  limit: string;
  offset: string;
  total: string;
  DriverTable: DriverTable;
}

export interface DriverTable {
  Drivers: Driver[];
}

export interface Driver {
  driverId: string;
  url: string;
  givenName: string;
  familyName: string;
  dateOfBirth: any;
  nationality: string;
  nameIcon: string;
}
