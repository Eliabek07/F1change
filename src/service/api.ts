import Axios from 'axios';

const api = Axios.create({
  baseURL: 'https://ergast.com/api/f1',
});

export default api;
