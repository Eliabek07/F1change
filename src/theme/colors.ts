export const color = {
  primary: {
    '50': '#F9FFE5',
    '100': '#EEFFC8',
    '200': '#D4FF79',
    '300': '#C4FB5B',
    '400': '#ABF229',
    '500': '#8CD80A',
    '600': '#6CAD03',
    '700': '#518308',
    '800': '#42670D',
    '900': '#385710',
    '950': '#1B3102',
  },
  neutro: {
    '50': '#FAFAF9',
    '100': '#F5F5F4',
    '200': '#E7E5E4',
    '300': '#D6D3D1',
    '400': '#A8A29E',
    '500': '#78716C',
    '600': '#57534E',
    '700': '#44403C',
    '800': '#292524',
    '900': '#1C1917',
    '950': '#0C0A09',
  },
  alert: {
    success: {
      '200': '#9AF4C3',
      '800': '#03623F',
    },
    info: {
      '200': '#CAD9FB',
      '800': '#373C9C',
    },
    warning: {
      '200': '#FADFA8',
      '800': '#944818',
    },
    error: {
      '200': '#F7C3C0',
      '800': '#8D2D27',
    },
  },
  background: {
    white: '#F7F7F7',
    'off-white': '#F9F9F9',
  },
};
