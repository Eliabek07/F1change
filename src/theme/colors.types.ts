export interface Primary {
  '50': string;
  '100': string;
  '200': string;
  '300': string;
  '400': string;
  '500': string;
  '600': string;
  '700': string;
  '800': string;
  '900': string;
  '950': string;
}

export interface Neutro {
  '50': string;
  '100': string;
  '200': string;
  '300': string;
  '400': string;
  '500': string;
  '600': string;
  '700': string;
  '800': string;
  '900': string;
  '950': string;
}

export interface Alert {
  success: Success;
  info: Info;
  warning: Warning;
  error: Error;
}

export interface Success {
  '200': string;
  '800': string;
}

export interface Info {
  '200': string;
  '800': string;
}

export interface Warning {
  '200': string;
  '800': string;
}

export interface Error {
  '200': string;
  '800': string;
}

export interface Background {
  white: string;
  'off-white': string;
}

export interface RootColors {
  primary: Primary;
  neutro: Neutro;
  alert: Alert;
  background: Background;
}
