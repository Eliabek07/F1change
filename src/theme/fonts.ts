export const fonts = {
  h1: {
    fontSize: 64,
    lineHeight: 76,
    fontFamily: 'Inter-Bold',
  },
  h2: {
    fontSize: 40,
    lineHeight: 48,
    fontFamily: 'Inter-Bold',
  },
  h3: {
    fontSize: 32,
    lineHeight: 38,
    fontFamily: 'Inter-Bold',
  },
  h4: {
    fontSize: 24,
    lineHeight: 28,
    fontFamily: 'Inter-Bold',
  },
  h5: {
    fontSize: 20,
    lineHeight: 26,
    fontFamily: 'Inter-Bold',
  },
  h6: {
    fontSize: 18,
    lineHeight: 28,
    fontFamily: 'Inter-Bold',
  },
  '16px': {
    bold: {
      fontSize: 16,
      lineHeight: 24,
      fontFamily: 'Inter-Bold',
    },
    semi_bold: {
      fontSize: 16,
      lineHeight: 24,
      fontFamily: 'Inter-SemiBold',
    },
    medium: {
      fontSize: 16,
      lineHeight: 24,
      fontFamily: 'Inter-Medium',
    },
    regular: {
      fontSize: 16,
      lineHeight: 24,
      fontFamily: 'Inter-Regular',
    },
  },
  '14px': {
    bold: {
      fontSize: 14,
      lineHeight: 20,
      fontFamily: 'Inter-Bold',
    },
    semi_bold: {
      fontSize: 14,
      lineHeight: 20,
      fontFamily: 'Inter-SemiBold',
    },
    medium: {
      fontSize: 14,
      lineHeight: 20,
      fontFamily: 'Inter-Medium',
    },
    regular: {
      fontSize: 14,
      lineHeight: 20,
      fontFamily: 'Inter-Regular',
    },
  },
  '12px': {
    bold: {
      fontSize: 12,
      lineHeight: 16,
      fontFamily: 'Inter-Bold',
    },
    semi_bold: {
      fontSize: 12,
      lineHeight: 16,
      fontFamily: 'Inter-SemiBold',
    },
    medium: {
      fontSize: 12,
      lineHeight: 16,
      fontFamily: 'Inter-Medium',
    },
    regular: {
      fontSize: 12,
      lineHeight: 16,
      fontFamily: 'Inter-Regular',
    },
  },
  button: {
    large: {
      fontSize: 16,
      lineHeight: 20,
      fontFamily: 'Inter-SemiBold',
    },
    medium: {
      fontSize: 14,
      lineHeight: 20,
      fontFamily: 'Inter-SemiBold',
    },
    small: {
      fontSize: 12,
      lineHeight: 20,
      fontFamily: 'Inter-SemiBold',
    },
  },
  links: {
    large: {
      fontSize: 16,
      lineHeight: 20,
      fontFamily: 'Inter-SemiBold',
    },
    medium: {
      fontSize: 14,
      lineHeight: 20,
      fontFamily: 'Inter-SemiBold',
    },
  },
};
