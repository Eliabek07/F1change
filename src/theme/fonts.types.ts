export interface H1 {
  fontSize: number;
  lineHeight: number;
  fontFamily: string;
}

export interface H2 {
  fontSize: number;
  lineHeight: number;
  fontFamily: string;
}

export interface H3 {
  fontSize: number;
  lineHeight: number;
  fontFamily: string;
}

export interface H4 {
  fontSize: number;
  lineHeight: number;
  fontFamily: string;
}

export interface H5 {
  fontSize: number;
  lineHeight: number;
  fontFamily: string;
}

export interface H6 {
  fontSize: number;
  lineHeight: number;
  fontFamily: string;
}

export interface N16px {
  bold: Bold;
  semi_bold: SemiBold;
  medium: Medium;
  regular: Regular;
}

export interface Bold {
  fontSize: number;
  lineHeight: number;
  fontFamily: string;
}

export interface SemiBold {
  fontSize: number;
  lineHeight: number;
  fontFamily: string;
}

export interface Medium {
  fontSize: number;
  lineHeight: number;
  fontFamily: string;
}

export interface Regular {
  fontSize: number;
  lineHeight: number;
  fontFamily: string;
}

export interface N14px {
  bold: Bold2;
  semi_bold: SemiBold2;
  medium: Medium2;
  regular: Regular2;
}

export interface Bold2 {
  fontSize: number;
  lineHeight: number;
  fontFamily: string;
}

export interface SemiBold2 {
  fontSize: number;
  lineHeight: number;
  fontFamily: string;
}

export interface Medium2 {
  fontSize: number;
  lineHeight: number;
  fontFamily: string;
}

export interface Regular2 {
  fontSize: number;
  lineHeight: number;
  fontFamily: string;
}

export interface N12px {
  bold: Bold3;
  semi_bold: SemiBold3;
  medium: Medium3;
  regular: Regular3;
}

export interface Bold3 {
  fontSize: number;
  lineHeight: number;
  fontFamily: string;
}

export interface SemiBold3 {
  fontSize: number;
  lineHeight: number;
  fontFamily: string;
}

export interface Medium3 {
  fontSize: number;
  lineHeight: number;
  fontFamily: string;
}

export interface Regular3 {
  fontSize: number;
  lineHeight: number;
  fontFamily: string;
}

export interface Button {
  large: Large;
  medium: Medium4;
  small: Small;
}

export interface Large {
  fontSize: number;
  lineHeight: number;
  fontFamily: string;
}

export interface Medium4 {
  fontSize: number;
  lineHeight: number;
  fontFamily: string;
}

export interface Small {
  fontSize: number;
  lineHeight: number;
  fontFamily: string;
}

export interface Links {
  large: Large2;
  medium: Medium5;
}

export interface Large2 {
  fontSize: number;
  lineHeight: number;
  fontFamily: string;
}

export interface Medium5 {
  fontSize: number;
  lineHeight: number;
  fontFamily: string;
}

export interface RootFonts {
  h1: H1;
  h2: H2;
  h3: H3;
  h4: H4;
  h5: H5;
  h6: H6;
  '16px': N16px;
  '14px': N14px;
  '12px': N12px;
  button: Button;
  links: Links;
}
