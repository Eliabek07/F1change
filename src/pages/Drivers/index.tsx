import React, {useEffect, useState} from 'react';
import {FlatList, View, TouchableOpacity} from 'react-native';
import api from '../../service/api';
import {useTheme} from '@emotion/react';
import Typography from '../components/Typography';
import {format, isBefore} from 'date-fns';
import {ptBR} from 'date-fns/locale';
import {Driver, Root} from './types';
import American from '../../assets/iconsNatiolality/american.svg';
import Belgium from '../../assets/iconsNatiolality/belgian.svg';
import British from '../../assets/iconsNatiolality/british.svg';
import French from '../../assets/iconsNatiolality/french.svg';
import Germany from '../../assets/iconsNatiolality/germany.svg';
import Italian from '../../assets/iconsNatiolality/italian.svg';
import New_zealander from '../../assets/iconsNatiolality/new_zealander.svg';
import Spain from '../../assets/iconsNatiolality/spainish.svg';
import Swedish from '../../assets/iconsNatiolality/swedish.svg';
import Thai from '../../assets/iconsNatiolality/thai.svg';
import Dutch from '../../assets/iconsNatiolality/dutch.svg';
import MaterialIcons from 'react-native-vector-icons/MaterialIcons';
import {useNavigation} from '@react-navigation/native';

const Drivers: React.FC = () => {
  const [data, setDate] = useState<Driver[] | []>([]);
  const {color} = useTheme();
  const {goBack} = useNavigation();
  const listDrivers = async () => {
    try {
      const {
        data: {
          MRData: {
            DriverTable: {Drivers: Data = []},
          },
        },
      }: {data: Root} = await api.get('/drivers.json');

      if (Data.length > 0) {
        const sort: Driver[] = Data?.filter(
          list => !isBefore(list.dateOfBirth, '1960-01-01'),
        )
          ?.sort(
            (a: any, b: any) =>
              //@ts-ignore
              new Date(a.dateOfBirth) - new Date(b.dateOfBirth),
          )
          .map(item => ({
            ...item,
            nameIcon: item.nationality.replace(/\s/, '_').toLowerCase(),
          }));
        setDate(sort || []);
      }
    } catch (error) {
      //
    }
  };

  const returnIcon = (nameIcon: string): any => {
    try {
      const nationalities: any = {
        american: <American />,
        belgian: <Belgium />,
        british: <British />,
        french: <French />,
        germany: <Germany />,
        italian: <Italian />,
        new_zealander: <New_zealander />,
        spanish: <Spain />,
        swedish: <Swedish />,
        thai: <Thai />,
        dutch: <Dutch />,
      };
      return nationalities[nameIcon];
    } catch (error) {
      return null;
    }
  };
  // };

  useEffect(() => {
    listDrivers();
  }, []);

  const header = () => {
    return (
      <View style={{padding: 20}}>
        <View style={{flexDirection: 'row', alignItems: 'center'}}>
          <View style={{flex: 0.5}}>
            <TouchableOpacity
              onPress={() => {
                goBack();
              }}
              style={{
                width: 40,
                padding: 12,
                borderRadius: 8,
                backgroundColor: color.primary[100],
                justifyContent: 'center',
                alignItems: 'center',
              }}>
              <MaterialIcons color={color.primary[950]} name="arrow-back-ios" />
            </TouchableOpacity>
          </View>
          <Typography
            style={{alignSelf: 'center'}}
            variant="14_semi_bold"
            color="neutro_900">
            História
          </Typography>
        </View>
        <View style={{marginTop: 20}}>
          <Typography variant="h5" color="neutro_950">
            História da F1
          </Typography>
          <Typography variant="16_semi_bold" color="neutro_500">
            Conheça os nomes que fizeram história, desde os pioneiros até os
            campeões inesquecíveis.
          </Typography>
        </View>
      </View>
    );
  };
  return (
    <View
      style={{
        flex: 1,
        backgroundColor: color.background.white,
      }}>
      {header()}

      <FlatList
        data={data}
        ListHeaderComponent={() => (
          <Typography variant="12_semi_bold" color="neutro_950">
            Pilotos
          </Typography>
        )}
        contentContainerStyle={{
          marginHorizontal: 16,
          gap: 10,
          backgroundColor: color.background.white,
          paddingBottom: 100,
        }}
        keyExtractor={item => item.driverId}
        renderItem={({item}) => (
          <View
            style={{
              borderWidth: 1,
              padding: 12,
              borderColor: color.neutro[200],
              flexDirection: 'row',
              borderRadius: 8,
              gap: 12,
              alignItems: 'center',
            }}>
            {returnIcon(item.nameIcon)}

            <View>
              <Typography variant="16_bold" color="neutro_950">
                {item.givenName} {item.nameIcon}
              </Typography>
              <Typography variant="14_medium" color="neutro_500">
                {format(
                  new Date(item.dateOfBirth),
                  "'Nascido em' dd 'de' MMMM 'de' yyyy",
                  {locale: ptBR},
                )}
              </Typography>
            </View>
          </View>
        )}
      />
    </View>
  );
};

export default Drivers;
