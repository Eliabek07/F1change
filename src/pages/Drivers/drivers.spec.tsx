import React from 'react';
import {render} from '@testing-library/react-native';
import Drivers from '.';
import {ThemeProvider} from '@emotion/react';
import {color} from '../../theme/colors';
import {fonts} from '../../theme/fonts';
import api from '../../service/api';

jest.mock('react-native-vector-icons/MaterialIcons', () => 'Icon');
jest.mock('@react-navigation/native', () => ({
  useNavigation: () => ({
    goBack: jest.fn(),
  }),
}));
jest.mock('react-native-gesture-handler', () => {});
jest.mock('../../service/api');
let axiosMocked: jest.Mocked<typeof api>;

describe('Testing Typography component', () => {
  beforeEach(() => {
    axiosMocked = api as jest.Mocked<typeof api>;
  });

  afterEach(() => {
    axiosMocked.get.mockClear();
  });
  it('Shoud render Typography', () => {
    const {toJSON} = render(
      <ThemeProvider
        theme={{
          color: color,
          fonts: fonts,
        }}>
        <Drivers />,
      </ThemeProvider>,
    );
    expect(toJSON()).toMatchSnapshot();
  });
});
