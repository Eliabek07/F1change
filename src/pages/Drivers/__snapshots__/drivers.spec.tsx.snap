// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Typography component Shoud render Typography 1`] = `
[
  <View
    style={
      {
        "backgroundColor": "#F9F9F9",
        "flex": 1,
      }
    }
  >
    <View
      style={
        {
          "padding": 20,
        }
      }
    >
      <View
        style={
          {
            "alignItems": "center",
            "flexDirection": "row",
          }
        }
      >
        <View
          style={
            {
              "flex": 0.5,
            }
          }
        >
          <View
            accessibilityState={
              {
                "busy": undefined,
                "checked": undefined,
                "disabled": undefined,
                "expanded": undefined,
                "selected": undefined,
              }
            }
            accessibilityValue={
              {
                "max": undefined,
                "min": undefined,
                "now": undefined,
                "text": undefined,
              }
            }
            accessible={true}
            collapsable={false}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              {
                "alignItems": "center",
                "backgroundColor": "#EEFFC8",
                "borderRadius": 8,
                "justifyContent": "center",
                "opacity": 1,
                "padding": 12,
                "width": 40,
              }
            }
          >
            <Icon
              color="#1B3102"
              name="arrow-back-ios"
            />
          </View>
        </View>
        <Text
          style={
            [
              {
                "fontFamily": "Inter-SemiBold",
                "fontSize": 14,
                "lineHeight": 20,
              },
              {
                "color": "#03623F",
              },
              {
                "alignSelf": "center",
                "color": "#03623F",
              },
            ]
          }
        >
          História
        </Text>
      </View>
      <View
        style={
          {
            "marginTop": 20,
          }
        }
      >
        <Text
          style={
            [
              {
                "fontFamily": "Inter-Bold",
                "fontSize": 20,
                "lineHeight": 26,
              },
              {
                "color": "#0C0A09",
              },
              {},
            ]
          }
        >
          História da F1
        </Text>
        <Text
          style={
            [
              {
                "fontFamily": "Inter-SemiBold",
                "fontSize": 16,
                "lineHeight": 24,
              },
              {
                "color": "#78716C",
              },
              {},
            ]
          }
        >
          Conheça os nomes que fizeram história, desde os pioneiros até os campeões inesquecíveis.
        </Text>
      </View>
    </View>
    <RCTScrollView
      ListHeaderComponent={[Function]}
      contentContainerStyle={
        {
          "backgroundColor": "#F9F9F9",
          "gap": 10,
          "marginHorizontal": 16,
          "paddingBottom": 100,
        }
      }
      data={[]}
      getItem={[Function]}
      getItemCount={[Function]}
      keyExtractor={[Function]}
      onContentSizeChange={[Function]}
      onLayout={[Function]}
      onMomentumScrollBegin={[Function]}
      onMomentumScrollEnd={[Function]}
      onScroll={[Function]}
      onScrollBeginDrag={[Function]}
      onScrollEndDrag={[Function]}
      removeClippedSubviews={false}
      renderItem={[Function]}
      scrollEventThrottle={0.0001}
      stickyHeaderIndices={[]}
      viewabilityConfigCallbackPairs={[]}
    >
      <View>
        <View
          collapsable={false}
          onLayout={[Function]}
        >
          <Text
            style={
              [
                {
                  "fontFamily": "Inter-SemiBold",
                  "fontSize": 12,
                  "lineHeight": 16,
                },
                {
                  "color": "#0C0A09",
                },
                {},
              ]
            }
          >
            Pilotos
          </Text>
        </View>
      </View>
    </RCTScrollView>
  </View>,
  ",",
]
`;
